isisd/isis_lsp.c

```diff
@@ -1313,6 +1316,64 @@ static void lsp_build(struct isis_lsp *lsp, struct isis_area *area)
 				lsp_build_internal_reach_ipv6(lsp, area, ipv6,
 							      metric);
 		}
+		char prefixbuf_1[PREFIX2STR_BUFFER];
+		char prefixbuf_2[PREFIX2STR_BUFFER];
+
+                struct isis_adjacency* adj;
+                bool flag = false;
+                for (ALL_LIST_ELEMENTS_RO(area->adjacency_list, node, adj)) {
+                        if (adj->level == IS_LEVEL_2)
+                                flag = true;
+                }
+
+                if(flag && (lsp->level == IS_LEVEL_2) && (area->is_type == 3) && (lsp->hdr.lsp_bits != LSPBIT_ATT)) {
+                        struct isis_lsp* lsp_tmp_ip;
+                        struct isis_lsp* lsp_tmp_ipv6;
+                        struct lspdb_head *head_tmp_ip = &area->lspdb[0];
+                        struct lspdb_head *head_tmp_ipv6 = &area->lspdb[0];
```

Comment:
Please use same variables for both IP and IPv6. There are temporary pointer and there is no reason to duplicate them.